tests/scripts/components-configuration-tls.sh

```diff
@@ -0,0 +1,835 @@
+# components-configuration-tls.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+# This file contains the test components that are executed by all.sh
+
+################################################################
+#### Configuration Testing - TLS
+################################################################
+
+component_test_no_renegotiation () {
+    msg "build: Default + !MBEDTLS_SSL_RENEGOTIATION (ASan build)" # ~ 6 min
+    scripts/config.py unset MBEDTLS_SSL_RENEGOTIATION
+    CC=$ASAN_CC cmake -D CMAKE_BUILD_TYPE:String=Asan .
+    make
+
+    msg "test: !MBEDTLS_SSL_RENEGOTIATION - main suites (inc. selftests) (ASan build)" # ~ 50s
+    make test
+
+    msg "test: !MBEDTLS_SSL_RENEGOTIATION - ssl-opt.sh (ASan build)" # ~ 6 min
+    tests/ssl-opt.sh
+}
+
+component_test_no_pem_no_fs () {
```

Comment:
`component_test_no_pem_no_fs` tests crypto options. It should go into the configurations-crypto group.

Also applies to `component_test_rsa_no_crt`, `component_test_*drbg*`, `component_test_everest*`, `component_full_without_ecdhe_ecdsa`.

Reply:
I do not think that moving `component_full_without_ecdhe_ecdsa` to configurations-crypto makes much sense since it is betters to be tested alongside `component_full_without_ecdhe_ecdsa_and_tls13 `

Everything else has been moved as requested.

~~~~~~
# Common helper for component_full_without_ecdhe_ecdsa() and
# component_full_without_ecdhe_ecdsa_and_tls13() which:
# - starts from the "full" configuration minus the list of symbols passed in
#   as 1st parameter
# - build
# - test only TLS (i.e. test_suite_tls and ssl-opt)
build_full_minus_something_and_test_tls () {

}

component_full_without_ecdhe_ecdsa () {
    build_full_minus_something_and_test_tls "MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED"
}

component_full_without_ecdhe_ecdsa_and_tls13 () {
    build_full_minus_something_and_test_tls "MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
                                             MBEDTLS_SSL_PROTO_TLS1_3"
}
~~~~~~

Reply:
> I do not think that moving component_full_without_ecdhe_ecdsa to configurations-crypto makes much sense

If this component doesn't become obsolete before the repo split, it'll go into the crypto repository, whereas the `and_tls13` variant will go into the TLS repository. That's why it's in the wrong place.

That does mean we'll need an auxiliary function that's used in two different files, and, down the line, in two different repos. @mpg Do you think `component_full_without_ecdhe_ecdsa` will be obsolete after we reach PSA-only configuration?

Reply:
Sorry @gilles-peskine-arm I don't understand why you think `component_full_without_ecdhe_ecdsa` would go to the crypto side: it's about a TLS option. We're testing the full config minus the TLS 1.2 key exchange ECDHE-ECDSA. That's quite definitely a TLS config thing.

Note: it's been a long-standing break of our own internal namespacing rules that TLS 1.2 key exchange options (and same for the enum values) are called `MBEDTLS_KEY_EXCHANGE_xxx` and not `MBEDTLS_SSL_KEY_EXCHANGE_xxx` as they should.

Reply:
Ah, my bad, I was only looking at the name and I hadn't noticed that this was only about key exchanges, not about algorithm support. So I agree that `component_full_without_ecdhe_ecdsa` belongs in TLS.